scipy/stats/_axis_nan_policy.py

```diff
@@ -515,37 +515,33 @@ def hypotest_fun(x):
                     if sentinel:
                         samples = _remove_sentinel(samples, paired, sentinel)
                     if is_too_small(samples):
-                        res = np.full(n_outputs, np.nan)
-                        return tuple_to_result(*res)
-                    return hypotest_fun_out(*samples, **kwds)
+                        return np.full(n_outputs, np.nan)
+                    return result_to_tuple(hypotest_fun_out(*samples, **kwds))
```

Comment:
Before, these calls to the 1D version of the function (which acts along a single slice) were returning a result object. You can see this because here we turn the `nan`s into an object like:
```python3
                        res = np.full(n_outputs, np.nan)
                        return tuple_to_result(*res)  # turn into a result object
```
or we simply call the `hypotest_fun_out`, which returns a result object. The reuslt object could be a tuple, but it could also be a more general object. 

Below, `apply_along_axis` called this 1D version of the function. If the result object was a tuple, it could compile the tuples from all the slices into a regular array. If the result object was a non-iterable object, it would make an array of result objects, and the old `result_unpacker` function would define how to deal with this. But if the result was an iterable object with extra fields, like a bunch, it treated it just like a normal tuple. It ignored the extra fields, and we lost some information.

In retrospect, the solution is an unambiguous improvement: these calls to the 1D version of the function should always return a simple tuple with all the information that is needed. We don't need to call `tuple_to_result` to turn the `NaN`s into a result object; we just return the sequence of NaNs. We call `result_to_tuple` to unpack more general objects - including all the information we need - into simple tuples. This way, `apply_along_axis` can always compile the results into a big non-object array.

Reply:
> In retrospect, the solution is an unambiguous improvement: these calls to the 1D version of the function should always return a simple tuple with all the information that is needed. We don't need to call `tuple_to_result` to turn the `NaN`s into a result object; we just return the sequence of NaNs. We call `result_to_tuple` to unpack more general objects - including all the information we need - into simple tuples. This way, `apply_along_axis` can always compile the results into a big non-object array.

Makes sense. I tested this and it solves the problem for Bunch objects.